[Security] Deprecate `UserInterface` & `TokenInterface`'s `eraseCredentials()` (#59106)
symfony/symfony: src/Symfony/Component/Security/Http/CHANGELOG.md

---

 * Add encryption support to `OidcTokenHandler` (JWE)

>  * Deprecate passing `true` for the `$eraseCredentials` parameter of `AuthenticatorManager::__construct()`, erase credentials
>   on your own e.g. upon `AuthenticationTokenCreatedEvent` instead. Passing it won't thave any effect in 8.0.

Reviewer: ```suggestion
 * Deprecate passing `true` for the `$eraseCredentials` parameter of `AuthenticatorManager::__construct()`,
    The value of this argument will be ignored in 8.0.
```
